symphony/subproc/manager.py

```diff
@@ -74,7 +74,8 @@ def launch(self, name, cmd, env):
             universal_newlines=True,  # force text stream
             stdout=stdout,
             stderr=stderr,
-            env=env
+            env=env,
+            preexec_fn=os.setsid
```

Comment:
I don't understand this line, why do you add `setsid`?

Reply:
This makes the subprocess have its own process group, which means that its children (so the grandchildren of the original process) will belong to the subprocess' process group, so that they react to the signals sent to the subprocsss. Otherwise the replay_worker thread became an orphan.